Build failure: thin-provisioning-tools

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Build this package for aarch64 on a x86_64 build platform:
```
nixpkgs.hostPlatform = "aarch64-linux";
nixpkgs.buildPlatform = "x86_64-linux";
```

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/4i96cn5fj6px9gxiwcf29d8yac0p7b1n-thin-provisioning-tools-aarch64-unknown-linux-gnu-1.2.0.drv' failed with exit code 101;
       last 25 log lines:
       >    Compiling udev v0.9.3
       >    Compiling rand v0.9.1
       >    Compiling iovec v0.1.4
       >    Compiling num_cpus v1.17.0
       >    Compiling quick-xml v0.37.5
       >    Compiling nom v8.0.0
       >    Compiling exitcode v1.1.2
       >    Compiling rangemap v1.5.1
       >    Compiling base64 v0.22.1
       >    Compiling data-encoding v2.9.0
       >    Compiling fixedbitset v0.5.7
       >    Compiling devicemapper-sys v0.3.2
       >    Compiling devicemapper v0.34.4
       > error: linking with `/nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/cc` failed: exit status: 1
       >   |
       >   = note:  "/nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/cc" "-m64" "/build/rustcqi6AJT/symbols.o" "<2 object files omitted>" "-Wl,--as-needed" "-Wl,-Bstatic" "/build/source/target/release/deps/{libdevicemapper_sys-cf60e939cfd64e09.rlib,libsemver-18fc9fd14d824b04.rlib}.rlib" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib/{libstd-*,libpanic_unwind-*,libobject-*,libmemchr-*,libaddr2line-*,libgimli-*,librustc_demangle-*,libstd_detect-*,libhashbrown-*,librustc_std_workspace_alloc-*,libminiz_oxide-*,libadler2-*,libunwind-*,libcfg_if-*,liblibc-*,liballoc-*,librustc_std_workspace_core-*,libcore-*,libcompiler_builtins-*}.rlib" "-Wl,-Bdynamic" "-ldevmapper" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-L" "/build/rustcqi6AJT/raw-dylibs" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/nix/store/rfznwwiqgpjrpjd9593c0fz7nbbrvnl5-lvm2-aarch64-unknown-linux-gnu-2.03.32-lib/lib" "-L" "<sysroot>/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/build/source/target/release/build/devicemapper-3cc823516cf7be0f/build_script_build-3cc823516cf7be0f" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
       >   = note: some arguments are omitted. use `--verbose` to show all linker arguments
       >   = note: /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: skipping incompatible /nix/store/rfznwwiqgpjrpjd9593c0fz7nbbrvnl5-lvm2-aarch64-unknown-linux-gnu-2.03.32-lib/lib/libdevmapper.so when searching for -ldevmapper
       >           /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: cannot find -ldevmapper: No such file or directory
       >           /nix/store/z3za8hfc24wb117s50p8b10agjkgm039-binutils-2.44/bin/ld: skipping incompatible /nix/store/rfznwwiqgpjrpjd9593c0fz7nbbrvnl5-lvm2-aarch64-unknown-linux-gnu-2.03.32-lib/lib/libdevmapper.so when searching for -ldevmapper
       >           collect2: error: ld returned 1 exit status
       >           
       >
       > error: could not compile `devicemapper` (build script) due to 1 previous error
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/4i96cn5fj6px9gxiwcf29d8yac0p7b1n-thin-provisioning-tools-aarch64-unknown-linux-gnu-1.2.0.drv'.
error: 1 dependencies of derivation '/nix/store/zdzdzn9p9md4k4xv3gmj4gbhcbs7wy34-libblockdev-aarch64-unknown-linux-gnu-3.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/zmjbhsra6hn717vnph5b4lnfvih9gd77-etc.drv' failed to build
error (ignored): error: cannot unlink 'ldb-2.9.2' in directory '/nix/var/nix/builds/nix-build-ldb-aarch64-unknown-linux-gnu-2.9.2.drv-0': Directory not empty
error (ignored): error: cannot unlink 'source' in directory '/nix/var/nix/builds/nix-build-imagemagick-7.1.1-47.drv-0': Directory not empty
error (ignored): error: cannot unlink 'nix-build-linux-aarch64-unknown-linux-gnu-6.16-rc2.drv-1' in directory '/nix/var/nix/builds': Directory not empty
error: 1 dependencies of derivation '/nix/store/n6pm0qz4ns1cci2nafkqvycd8qmcyq8p-udisks-aarch64-unknown-linux-gnu-2.10.1.drv' failed to build
error (ignored): error: cannot unlink 'dbus' in directory '/nix/var/nix/builds/nix-build-switch-to-configuration-aarch64-unknown-linux-gnu-0.1.0.drv-0/cargo-vendor-dir/libdbus-sys-0.2.5/vendor': Directory not empty
error: 1 dependencies of derivation '/nix/store/ykggj31a03fcchsk2fbkcjrg3m6dsvsa-nixos-system-quadphone-25.11.20250706.1fd8bad.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ly6y0aqgqw0lx7nnbgsa645m2vnzn4yh-ukify.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sxlcn1gvp1hpsb511625zc92vf7zmihi-image.drv' failed to build
```

### Additional context

Cross compilation bug, I think.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.35, NixOS, 25.11 (Xantusia), 25.11.20250706.1fd8bad`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/quadradical/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/quadradical/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/quadradical/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/quadradical/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/quadradical/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/fqr8b622ld52fifi77h3bgamp1b5xzy9-gnome-settings-daemon-48.1/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/3j073h93daccd3sk5nvm241sdpnd1aqz-lix-2.91.3/share`
 - nixpkgs: `/nix/store/i5zbq5by44gz4w1v0kwdcv2d8vacqhd1-source`

### Notify maintainers


@TomaSajt (not maintainer but last to edit package)
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/th/thin-provisioning-tools/package.nix
+++ b/pkgs/by-name/th/thin-provisioning-tools/package.nix
@@ -7,7 +7,6 @@
   fetchFromGitHub,
   nixosTests,
 }:
-
 rustPlatform.buildRustPackage rec {
   pname = "thin-provisioning-tools";
   version = "1.2.0";
@@ -19,13 +18,18 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-gjsURDzA4LRTTgKZPzzTcvTdi1mXx4FkWmyoPcpdPfU=";
   };
 
-  nativeBuildInputs = [
+  strictDeps = true;
+  depsBuildBuild = [
     pkg-config
+    lvm2
+    udev
+  ];
+  nativeBuildInputs = [
     rustPlatform.bindgenHook
   ];
   buildInputs = [
-    udev
     lvm2
+    udev
   ];
 
   useFetchCargoVendor = true;

